tests/lib/rules/arrow-spacing.js

```diff
@@ -0,0 +1,323 @@
+/**
+ * @fileoverview Tests for arrow-spacing
+ * @author Jxck
+ * @copyright 2015 Jxck. All rights reserved.
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+//
+
+var eslint = require("../../../lib/eslint"),
+    ESLintTester = require("eslint-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var eslintTester = new ESLintTester(eslint);
+
+var valid = [
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "a => a",
+        "args": [ 1, { "after": true, "before": true } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "() => {}",
+        "args": [ 1, { "after": true, "before": true } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "(a) => {}",
+        "args": [ 1, { "after": true, "before": true } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "a=> a",
+        "args": [ 1, { "after": true, "before": false } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "()=> {}",
+        "args": [ 1, { "after": true, "before": false } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "(a)=> {}",
+        "args": [ 1, { "after": true, "before": false } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "a =>a",
+        "args": [ 1, { "after": false, "before": true } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "() =>{}",
+        "args": [ 1, { "after": false, "before": true } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "(a) =>{}",
+        "args": [ 1, { "after": false, "before": true } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "a=>a",
+        "args": [ 1, { "after": false, "before": false } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "()=>{}",
+        "args": [ 1, { "after": false, "before": false } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "(a)=>{}",
+        "args": [ 1, { "after": false, "before": false } ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "a => a",
+        "args": [ 1, {} ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "() => {}",
+        "args": [ 1, {} ]
+    },
+    {
+        "ecmaFeatures": { "arrowFunctions": true },
+        "code": "(a) => {}",
+        "args": [ 1, {} ]
+    }
+];
+
+
+var invalid = [
+    {
+        "code": "a=>a",
+        "args": [ 1, { "after": true, "before": true } ],
```

Comment:
Please use `options` instead of args. It's the same thing except you don't need to include the meaningless 1 at the start.